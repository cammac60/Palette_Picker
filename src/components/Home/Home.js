import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { addProject, getProjects } from '../../apiCalls.js';
import Chooser from '../Chooser/Chooser';
import './Home.scss';
import { BrowserRouter } from 'react-router-dom';


export const handleProjectSubmit = async projectHooks => {
  if (projectHooks.newProject) {
    try {
      const projectID = await addProject(projectHooks.newProject);
      const newProjectObj =  await getProjects(projectID.id);
      projectHooks.setProjects([...projectHooks.projects, newProjectObj]);
      projectHooks.setNewProject('');
    } catch(error) {
        console.log(error);
        projectHooks.setError('There was a problem creating your project.');
    }
  } else {
      projectHooks.setError('Please enter a name.');
  }
};

export const Home = props => {
  const [newProject, setNewProject] = useState('');
  const [error, setError] = useState('');
  const { user, setUser, projects, setProjects } = props;
  const addProjectHooks = {newProject, setNewProject, projects, setProjects, setError};
  const projectNodes = projects.map(project => {
    return (
      <div className="project-button" key={project.id} id={project.id}>
        <img className="project-icon"
          src={require("../../images/icons/pantone.svg")} alt="Project-Icon"/>
        <span className="project-name">{project.name}</span>
      </div>
    )
  });

  return (
    <div className="home">
      <nav className="home-nav">
        <h2>Palette Picker</h2>
        <BrowserRouter>
        <Link className="sign-out" to="/" onClick={() => setUser({})}>Sign Out</Link>
        </BrowserRouter>
      </nav>
      <section className="project-section">
        <div className="user-container">
          <img alt="Profile Icon" className="profile-icon"
            src={require("../../images/icons/user.svg")}/>
          <span className="username-label">{user}</span>
        </div>
        <div className="projects-container">
          <div className="projects-header">
            <h3>Your Projects:</h3>
            <span className="add-project-error">{error}</span>
            <div className="add-project-wrapper">
              <input className="add-project" type="text"
                placeholder="New Project" onChange={e => {
                setNewProject(e.target.value);
                setError('');
              }} value={newProject} maxLength="24"/>
              <img alt="Add Icon" className="add-icon" src={require("../../images/icons/add-button.svg")} onClick={() => handleProjectSubmit(addProjectHooks)}/>
            </div>
          </div>
          <div className="small-project-wrapper">
            {projectNodes}
          </div>
        </div>
      </section>
      <div className="divider"></div>
      <Chooser projects={projects}/>
    </div>
  )
}

export default Home;
