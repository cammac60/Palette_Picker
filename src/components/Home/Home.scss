@import '../../variables.scss';

.home {
  background: {
    attachment: fixed;
    color: rgba(0,0,0,0.5);
    image: url('../../images/wave2.png');
    position: center;
    repeat: no-repeat;
    size: cover;
  }
  background-blend-mode: darken;
  margin: 0;
  min-height: 100%;
  min-width: 100%;
  overflow-x: scroll;
}

.home-nav {
  display: flex;
  justify-content: space-between;
  margin: {
    left: 40px;
    right: 40px;
  }
  padding-top: 20px;
}

h2 {
  font: {
    family: $title-font;
    size: 2.5rem;
  }
  text-decoration: underline;
}

.sign-out {
  color: $light-text-color;
  font-size: 1.2rem;
  margin: {
    bottom: auto;
    top: auto;
  }
  &:hover {
    color: $medium-text-color;
    text-decoration: none;
  }
}

.profile-icon {
  color: $light-text-color;
  height: 150px;
  margin: none;
}

.user-container {
  width: 15%;
}

.project-section {
  display: flex;
  height: 450px;
  justify-content: space-between;
  margin: {
    left: 40px;
    right: 40px;
    top: 5%;
  }
}

.projects-container {
  background-color: rgba(255, 255, 255, 0.5);
  border-radius: 4px;
  color: $dark-text-color;
  overflow-x: scroll;
  padding-bottom: 20px;
  width: 75%;
}

.username-label {
  display: block;
  font-size: 2rem;
}

.projects-header {
  display: flex;
  height: 45px;
  justify-content: space-between;
  margin-top: 2%;
  padding: {
    left: 20px;
    right: 190px;
  }
}

.add-project-wrapper {
  display: flex;
  justify-content: space-between;
  width: 15%;
}

.add-icon {
  height: 100%;
  &:active {
    background: rgb(255,255,255);
    background: radial-gradient(circle, rgba(255, 255, 255, 1) 0%, rgba(255, 255,255, 0) 100%);
    border-radius: 50%;
  }
}

h3 {
  font-size: 2rem;
}

.add-project {
  font-size: 1.2rem;
  margin-right: 20px;
  padding: 5px;
}

.small-project-wrapper {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
  margin: {
    left: auto;
    right: auto;
    top: 40px;
  }
  width: 90%;
}

.project-button {
  border-radius: 4px;
  padding: 10px;
  &:hover {
    background-color: $light-text-color;
  }
  &:active {
    background-color: #FFF200;
  }
}

.project-icon {
  height: 100px;
}

.project-name {
  display: block;
  margin-top: 20px;
}

.divider {
  background-color: $light-text-color;
  min-height: 5px;
  margin: {
    bottom: 10%;
    left: auto;
    right: auto;
    top: 10%;
  }
  width: 95%;
}

.add-project-error {
  font-size: 1.2rem;
}
