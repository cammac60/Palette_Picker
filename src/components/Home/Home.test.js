import React from 'react';
import ReactDOM from 'react-dom';
import { Home, handleProjectSubmit } from './Home.js';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { BrowserRouter } from 'react-router-dom';



describe('Home', () => {
  let wrapper, mockUser, mockSetUser, mockSetProjects;
  const mockProjects = [
    {id: 1, name: 'Project 1'},
    {id: 2, name: 'Project 2'}
  ];
  const mockProps = {projects: mockProjects};

  it('Should match snapshot', () => {
    mockUser = 'Ben';
    mockSetUser = jest.fn();
    mockSetProjects = jest.fn();
    wrapper = shallow(<Home
      user={mockUser}
      setUser={mockSetUser}
      projects={mockProjects}
      setProjects={mockSetProjects}
    />);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should render without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Home  {...mockProps} />
      </BrowserRouter>
    , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should load with an initial project text of an empty string', () => {
    const { container } = render(
      <BrowserRouter>
        <Home {...mockProps}/>
      </BrowserRouter>
    );
    const project = container.querySelector('.add-project');
    expect(project.textContent).toBe('');
  });
});
