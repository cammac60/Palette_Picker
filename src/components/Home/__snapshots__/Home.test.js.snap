// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should match snapshot 1`] = `
<div
  className="home"
>
  <nav
    className="home-nav"
  >
    <h2>
      Palette Picker
    </h2>
    <BrowserRouter>
      <Link
        className="sign-out"
        onClick={[Function]}
        to="/"
      >
        Sign Out
      </Link>
    </BrowserRouter>
  </nav>
  <section
    className="project-section"
  >
    <div
      className="user-container"
    >
      <img
        alt="Profile Icon"
        className="profile-icon"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "user.svg",
          }
        }
      />
      <span
        className="username-label"
      >
        Ben
      </span>
    </div>
    <div
      className="projects-container"
    >
      <div
        className="projects-header"
      >
        <h3>
          Your Projects:
        </h3>
        <span
          className="add-project-error"
        />
        <div
          className="add-project-wrapper"
        >
          <input
            className="add-project"
            onChange={[Function]}
            placeholder="New Project"
            type="text"
            value=""
          />
          <img
            alt="Add Icon"
            className="add-icon"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "add-button.svg",
              }
            }
          />
        </div>
      </div>
      <div
        className="small-project-wrapper"
      >
        <div
          className="project-button"
          id={1}
          key="1"
        >
          <img
            alt="Project-Icon"
            className="project-icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "pantone.svg",
              }
            }
          />
          <span
            className="project-name"
          >
            Project 1
          </span>
        </div>
        <div
          className="project-button"
          id={2}
          key="2"
        >
          <img
            alt="Project-Icon"
            className="project-icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "pantone.svg",
              }
            }
          />
          <span
            className="project-name"
          >
            Project 2
          </span>
        </div>
      </div>
    </div>
  </section>
  <div
    className="divider"
  />
</div>
`;
